packages/fxa-auth-server/fxa-oauth-server/docs/pairwise-pseudonymous-identifiers.md

```diff
@@ -0,0 +1,68 @@
+
+### RP initiated PPID rotation
+
+Any RP that receives JWT access tokens or OIDC ID tokens can initiate PPID rotation by specifying an additional parameter, `ppid_seed`, when requesting tokens from either the `/authorization` or `/token` endpoints on the OAuth server. `ppid_seed` must be an integer between 0 and 1024, and can be rotated at any time.
+
+### Server initiated PPID rotation
+
+By default, Firefox accounts does not enforce sub rotation, but for sensitive RPs where long term user profiles are undesirable, Firefox accounts can enforce periodic rotation without depending on the RP to pass in a `ppid_seed`. The default period is 6 hours.
+
+## Ensuring 3rd parties are unable to access user information
+
+An OAuth access token is a bearer token that can be presented by anyone to an OAuth service provider to access a protected resource. If a JWT access token contains a PPID `sub` claim that is meant to protect a user's privacy, but is granted the `profile` scope allows the service provider to present the same access token to the FxA profile server and learn the user's true identity. As such, tokens _MUST NOT_ be requested with the `profile` scope or any of its implicants such as `profile:uid` or `profile:email`.
+
+## PPIDs and logging
+
+PPIDs are not used by internal Mozilla properties and are not logged.
+
+## PPID generation
+
+PPIDs are generated using [HKDF][#hkdf] with the following inputs:
+
+- **km**: `client_id`.`userid_hex`.`ppid_seed||0`.`serverEnforcedRotationInput`
+- **info**: `oidc ppid sub`
+- **salt**: A secret configured by Ops
+- **len**: 8 bytes, which leads to a 16 hex character result
+
+### Server enforced rotation input
+
+The server enforced rotation input defaults to the string `0`. If server enforced rotation is enabled for the client, it is calculated as follows:
+
+```
+serverEnforcedRotationInput = Math.floor(Date.now() / ROTATION_PERIOD)
+```
+
+If an RP requests two tokens in short succession, their `sub` claims could be different if the first token was generated previous to the rotation epoch and the second after.
+
+## The future
+
+Firefox accounts does not currently support [sector identifiers][#oidc-sector-identifier]. Sector identifiers allow multiple client_ids to receive the same PPID, this would be useful if the RP has applications on multiple platforms, e.g., an app on Android, iOS, and an addon in Firefox.
+
+[#oidc-spec]: https://openid.net/specs/oidc-core-1_0.html#Terminology
+[#oidc-id-token]: https://openid.net/specs/openid-connect-core-1_0.html#IDToken
+[#oidc-sector-identifier]: https://openid.net/specs/openid-connect-core-1_0.html#PairwiseAlg
+[#oauth-jwt-access-token]: https://tools.ietf.org/html/draft-bertocci-oauth-access-token-jwt-00
+[#hkdf]: https://tools.ietf.org/html/rfc5869
```

Comment:
Thank you for this doc! I'm sorry to leave so many nits, it is a very good doc overall :-)